care/users/models.py

```diff
@@ -363,3 +364,16 @@ def save(self, *args, **kwargs) -> None:
         if self.district is not None:
             self.state = self.district.state
         super().save(*args, **kwargs)
+
+
+class UserFacilityAllocation(models.Model):
+    """
+    This model tracks the allocation of a user to a facility for metabase analytics.
+    """
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="+")
+    facility = models.ForeignKey(
+        "facility.Facility", on_delete=models.CASCADE, related_name="+"
+    )
```

Comment:
```suggestion
    user = models.ForeignKey(User, on_delete=models.CASCADE)
    facility = models.ForeignKey(
        "facility.Facility", on_delete=models.CASCADE)
```

@gigincg should we allow backward relation so that we could do things like show a user allocation timeline graph in the care fe or list users in a facility for a specific time range?

Reply:
@gigincg bump!

Reply:
This is not planned currently